### PR TITLE
Show page not available for deleted poste urls

### PR DESCRIPTION
**Pull request by molly-moen**
Show a page not available message for poste urls that have been deleted by the script linked above. The page message is:
Screen Shot 2020-08-10 at 9 11 09 AM

## Links

- [jira](https://codedotorg.atlassian.net/browse/PLC-937)

## Testing story
Validated locally that a deleted url will show the page not available message while a url that has not deleted will still redirect as expected.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: pegasus/sites.v3/code.org/views/page_not_available.haml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
```
Comment (bencodeorg) [inline pegasus/sites.v3/code.org/views/page_not_available.haml]: You had to make this page from scratch?

File: pegasus/routes/poste_routes.rb [hunk 2]
```diff
@@ -16,6 +16,11 @@
   end
   url = DB[:poste_urls].where(id: url_id).first
   pass unless url
+  if url[:deleted_at]
+    path = resolve_template('views', settings.template_extnames, 'page_not_available')
+    content = path ? document(path) : "Sorry, this page is no longer available.\n"
```
Comment (bencodeorg) [inline pegasus/routes/poste_routes.rb:21]: When would `path` not exist?

Comment (molly-moen) [reply pegasus/routes/poste_routes.rb:21]: it's very unlikely (I think only if someone moved the haml file) but I'd rather have the check here than have the page give a 404 due to an exception

File: pegasus/sites.v3/code.org/views/page_not_available.haml (hunk 1 above)
Comment (molly-moen) [reply pegasus/sites.v3/code.org/views/page_not_available.haml]: oops, removed the unnecessary lines at the top

Comment (bencodeorg) [reply pegasus/sites.v3/code.org/views/page_not_available.haml]: Hah didn't even notice that, I just meant we don't have a page not available view already (looks like maybe not browsing through pegasus views...)

Comment (molly-moen) [reply pegasus/sites.v3/code.org/views/page_not_available.haml]: oh yeah we have a page not found but we want something more custom that points people to support if they think they need access

File: pegasus/sites.v3/code.org/views/page_not_available.haml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  @header["title"] = I18n.t(:page_not_available_title)
+
+%link{href: "/css/errors.css", rel: "stylesheet"}
+.error-parent
+  .error-child
+    %img{src: '/shared/images/sad-bee-avatar.png'}
+    %br
+    .error-message
+      %h1= I18n.t(:page_not_available_title)
+      = I18n.t(:page_not_available_description).html_safe
```
Comment (bencodeorg) [inline pegasus/sites.v3/code.org/views/page_not_available.haml]: @Hamms is this an example where we shouldn't insert HTML straight from a translator? [Here's a link to a Slack discussion where we discussed this in more detail.](https://codedotorg.slack.com/archives/C0T0UQH0R/p1594846525154300)

Comment (bencodeorg) [reply pegasus/sites.v3/code.org/views/page_not_available.haml]: Hmm Hamms is out this week, here's an example he helped me with:

https://github.com/code-dot-org/code-dot-org/blob/eb7a9f474743e2b2d84afe8bf3a14c5ae1ba11f9/dashboard/app/views/layouts/_thank_donors_interstitial.html.haml#L19

There's more context in the slack thread, and I just posted in i18n to see if anyone has any more information to share.

Comment (molly-moen) [reply pegasus/sites.v3/code.org/views/page_not_available.haml]: updated to use markdown instead!

**Review by islemaster — APPROVED** [review]
:tada: Looks great.

**Review by bencodeorg — APPROVED** [review]

**Pull request merged**